### PR TITLE
refactor normalize sequence

### PR DESCRIPTION
**Pull request by piiswrong**
@szha  dropout update

### CODE REVIEW

**Comment by loofahcus** [comment]
@piiswrong I think [cudnn_lstm_bucketing.py#L78-L86](https://github.com/dmlc/mxnet/blob/master/example/rnn/cudnn_lstm_bucketing.py#L78-L86) can be updated, too.

**Comment by piiswrong** [comment]
_ means protected function. pack/unpack are public

**Comment by szha** [comment]
Yeah, underscore prefix usually marks "internal use" properties or methods. So do we expect users to explicitly call pack_weights, or do we expect solely the "checkpoint" methods to call those. For the former, I'd recommend that we document all public APIs, while in later case we should add the underscore to make it clear it's not for regular users.

File: python/mxnet/rnn/rnn_cell.py [hunk 1]
```diff
@@ -931,17 +743,50 @@ def unroll(self, length, inputs=None, begin_state=None,
...
 
         return inputs, states
 
+
+class DropoutCell(BaseRNNCell):
+    """Apply dropout on base cell
+
+    Parameters
+    ----------
+    dropout : float
```
Comment (szha) [inline python/mxnet/rnn/rnn_cell.py]: Consider using a more informative name. tf seems to be using "keep_prob"

**Pull request merged**